share/functions/__fish_pwd.fish

```diff
@@ -1,10 +1,10 @@
 switch (uname)
     case 'CYGWIN_*'
         function __fish_pwd --description "Show current path"
-            pwd | sed -e 's-^/cygdrive/\(.\)/\?-\u\1:/-'
+            pwd | string replace -r '^/cygdrive/(.)?' '\\\U$1:'
```

Comment:
Unfortunately this is affected by #5474 - we run an additional round of unescaping on the replacement argument. I felt it was too risky to change, so I implemented the fix behind a feature flag (to be clear this means it *will* become the default and eventually only behavior - it's not a configuration option you can choose to taste). For the time being this needs to be

```fish
if status test-feature regex-easyesc
    pwd | string replace -r '^/cygdrive/(.)?' '\U$1:'
else
    pwd | string replace -r '^/cygdrive/(.)?' '\\\U$1:'
end
```

Reply:
Ahhhh... I had a hard time figuring that one out.

Thanks for the explanation.